plugins/libplugin.h

```diff
@@ -96,6 +98,17 @@ struct command_result *forward_result(struct command *cmd,
 				      const jsmntok_t *result,
 				      void *arg);
 
+/* Callback for timer where we expect a 'command_result'. */
+struct command_result *timer_complete(void);
```

Comment:
Needs more clarification: should this *always* be called (possibly with the result destroyed by the caller if the timer is being used for non-command reason)?

If timer callback does not call this, `in_timer` remains `true`, should we check after return of callback that `in_timer` is not false?

Reply:
I had a check, unfortunately this is not always called immediately; in particular, the timer cb for autoclean calls send_outreq().

I'll clarify the documentation, though I suspect the API will evolve as we get more users.